tests/agent_server/test_server_agent.py

```diff
@@ -719,6 +727,26 @@ async def test_get_set_param(resource_container, environment, client, server):
     assert result.code == 200
 
 
+async def test_register_setting(resource_container, environment, client, server):
+    """
+    Test getting and setting params
+    """
+    resource_container.Provider.reset()
```

Comment:
I don't think you need the `resource_container`.
```suggestion
async def test_register_setting(environment, client, server):
    """
    Test registering a new setting.
    """
```